src/components/catalogue/NumberInputComponent.svelte

```diff
@@ -57,10 +70,39 @@
             },
             $activeQueryGroupIndex,
         );
+
+        resetToEmptySearchBar();
+    }
+
+    function onkeydown(event: KeyboardEvent) {
+        if (inSearchBar === false) return;
+
+        if (event.key === "Enter") {
+            addItem();
+        }
+
+        if (event.key === "ArrowUp" || event.key === "ArrowDown") {
+            event.preventDefault();
+        }
+    }
+
+    function onfocusin(event: FocusEvent) {
+        if (!inSearchBar) return;
+        // toInput can not be reached by tab when the focus is outside the form,
+        // so this can handle the focus via mouse click instead of using another event listener
+        if (event.target === toInput) return;
+
+        const relatedTargetOutside =
+            event.relatedTarget instanceof Node &&
+            !form.contains(event.relatedTarget);
+
+        if (relatedTargetOutside) {
+            fromInput.focus();
+        }
```

Comment:
Instead of handling these in the input components could you try my suggestion of handling them completely from the search bar using capture phase event handlers (https://svelte.dev/tutorial/svelte/capturing)?

For the enter key you can expose `addItem` to the outside.

The arrow keys are already handled from the outside if you now also call `event.preventDefault()` during the capture phase the input won't see the arrow key event so no need to ignore it here.

I'm not 100% sure about the focus logic but it seems it is already partially handled outside. I think it would be fine if you drill down to the first input element with a smart query selector from the outside and just focus that.